### PR TITLE
fix: adjusted the timing of `UpgradeBuildInSystemContract`

### PR DESCRIPTION
**Pull request by Pythonberg1997**
### Description

This pr is to adjusted the timing of `UpgradeBuildInSystemContract`

### Rationale

To avoid set code too early

### Changes

Notable changes: 
* move the call to `UpgradeBuildInSystemContract` from the beginning of a block to the end of a block

### CODE REVIEW

File: core/state_processor.go [hunk 1]
```diff
@@ -73,12 +73,15 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
...
-	lastBlock := p.bc.GetBlockByHash(block.ParentHash())
-	if lastBlock == nil {
-		return statedb, nil, nil, 0, fmt.Errorf("could not get parent block")
+
+	if !p.config.IsFeynman(block.Number(), block.Time()) {
+		// Handle upgrade build-in system contract code
+		lastBlock := p.bc.GetBlockByHash(block.ParentHash())
+		if lastBlock == nil {
+			return statedb, nil, nil, 0, fmt.Errorf("could not get parent block")
+		}
```
Comment (zzzckck) [inline core/state_processor.go:82]: we can move line 79 -> line 82 out of the if scope, i.e. check lastBlock before Feynman check

File: core/state_processor.go [hunk 2]
```diff
@@ -131,6 +134,15 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
...
 
+	if p.config.IsFeynman(block.Number(), block.Time()) {
+		// Handle upgrade build-in system contract code
+		lastBlock := p.bc.GetBlockByHash(block.ParentHash())
+		if lastBlock == nil {
+			return statedb, nil, nil, 0, fmt.Errorf("could not get parent block")
+		}
+		systemcontracts.UpgradeBuildInSystemContract(p.config, blockNumber, lastBlock.Time(), block.Time(), statedb)
+	}
+
```
Comment (zjubfd) [inline core/state_processor.go:145]: You only change the code of block Processor, the miner need change too.

**Review by zjubfd — APPROVED** [review]

File: core/state_processor.go (hunk 1 above)
Comment (Pythonberg1997) [reply core/state_processor.go:82]: Done

**Review by buddh0 — APPROVED** [review]

File: eth/tracers/api.go [hunk 3]
```diff
@@ -610,18 +611,37 @@ func (api *API) traceBlock(ctx context.Context, block *types.Block, config *Trac
 			return api.traceBlockParallel(ctx, block, statedb, config)
 		}
 	}
+
```
Comment (zjubfd) [inline eth/tracers/api.go:614]: please check trace tx and traceBlockParallel function as well

**Review by zzzckck — APPROVED** [review]

**Pull request merged**